service/matching/taskQueueManager.go

```diff
@@ -505,9 +521,6 @@ func (c *taskQueueManagerImpl) renewLeaseWithRetry(fsb fatalSignalBehavior) (tas
 	err := backoff.Retry(op, persistenceOperationRetryPolicy, common.IsPersistenceTransientError)
 	if err != nil {
 		c.metricScope().IncCounter(metrics.LeaseFailurePerTaskQueueCounter)
-		if fsb == emitFatalSignal && c.errShouldUnload(err) {
```

Comment:
Are we no longer actively unload task queue when they fail to renewLease, and only rely on GC to unload them?

Reply:
The decision to unload has just been moved out to the callers of this function rather then implemented inline. Sometimes the callers want to signal a fatal error because they're on a background thread (rpc request handler or one of the IO threads) but sometimes they want to emit a synchronous error because (a) it's cleaner and (b) to do otherwise deadlocks (e.g. in the case of Start())